loader/loader.c

```diff
@@ -492,6 +492,12 @@ static bool loaderAddJsonEntry(const struct loader_instance *inst,
                                LPSTR json_path,    // JSON string to add to the list reg_data
                                DWORD json_size,    // size in bytes of json_path
                                VkResult *result) {
+
+    // Check for and ignore duplicates.
+    if (*reg_data && strstr(*reg_data, json_path)) {
+        return true;
```

Comment:
You are de-duplicating the path to json files. Because of the way that the loader-ICD interface operates, it seems like we should de-duplicate the path to ICDs. I thought that ICD-path de-duplication was already performed by the loader. Not sure if it was lost, is broken, or I'm hallucinating it.

Additionally de-duplicating the path to json files seems like a good idea.

Reply:
There is some logic to remove duplicates in the loader, but it looks like it doesn't work when the duplicates are both in the DriverStore. The logic is [here](https://github.com/KhronosGroup/Vulkan-Loader/blob/171f2a6b44546c8d89273233ac771f8c05f11f62/loader/loader.c#L825). It was really intended to solve the case where a driver was included in both the DriverStore and the old (`HKLM\Software\Khronos\Vulkan\Drivers`) registry path. As a result, I don't think it every got tested when the duplicates are in the driver store.

Reply:
Thanks for the link to the existing dedup code. OK, I see now that it has a different purpose.

After thinking about this more, I'm OK with deduplicating only the paths to json files in this PR.

Longer term, I'm wondering if it wouldn't be better for the loader/ICD interface to pass the AdapterLuid into the ICD when asking which vkPhysicalDevices it exposes, and requiring that the ICD return vkPhysicalDevices only for that AdapterLuid. This would mean that the loader asks the ICD N times when there are N AdapterLuids associated with a given ICD.  Debating the merit of this change to loader/ICD interface belongs in a separate github issue from the one motivating this PR.